server_admin/topics/vault.adoc

```diff
@@ -216,3 +216,161 @@ To install and use the above custom provider the configuration would look someth
+
+You create the credential store and a vault where the credential store and vault passwords are in plain text.
+
+.Prerequisites
+
+* A running LDAP instance has `BIND DN credential: secret12`.  
+
+* The alias uses the format <realm-name>_< key-value> when using the default key resolver. In this case, the instance is running in the realm `ldaptest` and `ldaptest_ldap_secret` is the alias that corresponds to the value `ldap_secret` in that realm.
+
+.Procedure
+
+. Create the credential store.
++
+[source,bash,subs=+attributes]
+----
+[standalone@localhost:9990 /] /subsystem=elytron/credential-store=test-store:add(create=true, location=/home/test/test-store.p12, credential-reference={clear-text=testpwd1!},implementation-properties={keyStoreType=PKCS12})
+----
+
+. Add an alias to the credential store. 
++
+[source,bash,subs=+attributes]
+----
+/subsystem=elytron/credential-store=test-store:add-alias(alias=ldaptest_ldap__secret,secret-value=secret12)
+----
+
+. List the aliases from the credential store.
++
+[source,bash,subs=+attributes]
+----
+keytool -list -keystore ./test-store.p12 -storetype PKCS12 -storepass testpwd1!
+Keystore type: PKCS12
+Keystore provider: SUN
+
+Your keystore contains 1 entries
+
+ldaptest_ldap__secret/passwordcredential/clear/, Oct 12, 2020, SecretKeyEntry, 
+----
+
+. Configure the vault SPI. 
++
+[source,bash,subs=+attributes]
+----
+/subsystem=keycloak-server/spi=vault:add(default-provider=elytron-cs-keystore)
+
+/subsystem=keycloak-server/spi=vault/provider=elytron-cs-keystore:add(enabled=true, properties={location=>/home/test/test-store.p12, secret=>testpwd1!, keyStoreType=>PKCS12})
+----
++
+At this point, the vault and credentials store passwords are not encrypted.
++
+[source,bash,subs=+attributes]
+----
+        <spi name="vault">
+                <default-provider>elytron-cs-keystore</default-provider>
+                <provider name="elytron-cs-keystore" enabled="true">
+                    <properties>
+                        <property name="location" value="/home/test/test-store.p12"/>
+                        <property name="secret" value="testpwd1!"/>
+                        <property name="keyStoreType" value="PKCS12"/>
+                    </properties>
+                </provider>
+            </spi>
+
+         <credential-stores>
+                <credential-store name="test-store" location="/home/test/test-store.p12" create="true">
+                    <implementation-properties>
+                        <property name="keyStoreType" value="PKCS12"/>
+                    </implementation-properties>
+                    <credential-reference clear-text="testpwd1!"/>
+                </credential-store>
+         </credential-stores>
+----
+
+. In the LDAP provider, replace `binDN credential` with `${vault.ldap_secret}`.
+
+. Test your LDAP connection.
++
+.LDAP Vault
+image:images/ldap-vault.png[LDAP Vault]
+
+
+==== Adding encryption to the credential store and vault
```

Comment:
What we do is we mask only the password that can open the Elytron credential store (which in this case is a Java KeyStore). No encryption involved.